### PR TITLE
OSDOCS#8476: Adding CLI docs for 4.15

### PR DESCRIPTION
**Pull request by bergerhoffer**
Version(s):
4.15

Issue:
https://issues.redhat.com/browse/OSDOCS-8476

Link to docs preview:
* https://71213--ocpdocs-pr.netlify.app/openshift-enterprise/latest/cli_reference/openshift_cli/developer-cli-commands
* https://71213--ocpdocs-pr.netlify.app/openshift-enterprise/latest/cli_reference/openshift_cli/administrator-cli-commands

QE review:
- [x] QE has approved this change.

Additional information:

**Note to peer reviewer: this is autogenerated content, so I can't manually change the wording. But I do plan to submit fixes to the eng repos with a few fixes, so feel free to point out anything you notice.**

### CODE REVIEW

File: modules/oc-by-example-content.adoc [hunk 1]
```diff
@@ -2209,18 +2240,6 @@ Run a proxy to the Kubernetes API server
 
 
 
-== oc registry info
```
Comment (bergerhoffer) [inline modules/oc-by-example-content.adoc:2212]: This command is being removed from this doc reference. I tested it, and when doing `oc registry --help`, `info` is listed in 4.14, but is not listed for 4.15. However, if I do `oc registry info --help`, it still does return with some values.

Do we know if this command is actually being removed in 4.15, or did something change that's not letting it appear if it should?

The help for it does say that it's experimental. But we have other experimental flags still show up in these docs (like `oc get-token` above in this PR).

Can anyone help confirm this? @ingvagabund @zhouying7780, or @stevsmit do you know anyone a registry SME who might know?

**Comment by zhouying7780** [comment]
sure, I'll review today

File: modules/oc-by-example-content.adoc (hunk 1 above)
Comment (ingvagabund) [reply modules/oc-by-example-content.adoc:2212]: `oc registry info` still exists and is not removed in 4.15

Comment (ingvagabund) [reply modules/oc-by-example-content.adoc:2212]: The command has always been experimental, never promoted to stable. Though, for the sake of letting people know and have them ready let's do it.

> And maybe that's why it got removed from the CLI docs. I don't remember for sure, but we might not include commands that are deprecated. Do you know?

Unfortunately, I don't know.

Comment (bergerhoffer) [reply modules/oc-by-example-content.adoc:2212]: @soltysh Any chance you remember if our CLI doc generation script excludes deprecated CLI commands?

Comment (bergerhoffer) [reply modules/oc-by-example-content.adoc:2212]: @ingvagabund Do you know if there's a replacement for `oc registry info` that should be used instead? I see a few places in our docs where we use it.

* https://github.com/openshift/openshift-docs/blob/main/modules/disabling-redirect-private-storage-endpoint-azure.adoc?plain=1#L36
* https://github.com/openshift/openshift-docs/blob/main/modules/configuring-private-storage-endpoint-azure-vnet-subnet-iro-discovery.adoc?plain=1#L65

Comment (ingvagabund) [reply modules/oc-by-example-content.adoc:2212]: The closest replacement is to run `oc get imagestream -n openshift` and check the `IMAGE REPOSITORY` column.

File: modules/oc-adm-by-example-content.adoc [hunk 2]
```diff
@@ -82,6 +82,17 @@ Deny a certificate signing request
 == oc adm copy-to-node
 Copies specified files to the node.
 
+.Example usage
+[source,bash,options="nowrap"]
+----
+  # copy a new bootstrap kubeconfig file to every node
+  oc adm copy-to-node --copy=new-bootstrap-kubeconfig=/etc/kubernetes/kubeconfig
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:89]: The command will failed with error :
`oc adm copy-to-node --copy=new-bootstrap-kubeconfig=/etc/kubernetes/kubeconfig-test
error: unable to get interesting nodes: You must provide one or more resources by argument or filename.
Example resource specifications include:
   '-f rsrc.yaml'
   '--filename=rsrc.json'
   '<resource> <name>'
   '<resource>'`

File: modules/oc-adm-by-example-content.adoc [hunk 3]
```diff
@@ -82,6 +82,17 @@ Deny a certificate signing request
...
+
+  # copy a new bootstrap kubeconfig file to masters
+  oc adm copy-to-node --copy=new-bootstrap-kubeconfig=/etc/kubernetes/kubeconfig -l node-role.kubernetes.io/master
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:92]: same here , the command will failed

File: modules/oc-adm-by-example-content.adoc [hunk 4]
```diff
@@ -365,16 +376,45 @@ Watch platform certificates.
 == oc adm ocp-certificates regenerate-leaf
 Regenerate client and serving certificates of an OpenShift cluster
 
+.Example usage
+[source,bash,options="nowrap"]
+----
+  # Regenerate all known leaf certificates on the cluster.
+  oc adm certificates regenerate-leaf -A --all
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:383]: we don't have such command `oc adm certificates regenerate-leaf`

File: modules/oc-adm-by-example-content.adoc [hunk 5]
```diff
@@ -365,16 +376,45 @@ Watch platform certificates.
...
+
+  # Regenerate a leaf certificate contained in a particular secret.
+  oc adm ocp-certificates regenerate-leaf -n openshift-config-managed kube-controller-manager-client-cert
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:386]: here should be `oc adm ocp-certificates regenerate-leaf -n openshift-config-managed secret/kube-controller-manager-client-cert-key`

File: modules/oc-adm-by-example-content.adoc [hunk 6]
```diff
@@ -365,16 +376,45 @@ Watch platform certificates.
...
 
 
 == oc adm ocp-certificates regenerate-machine-config-server-serving-cert
 Regenerate the machine config operator certificates in an OpenShift cluster
 
+.Example usage
+[source,bash,options="nowrap"]
+----
+  # Regenerate the MCO certs without modifying user-data secrets
+  oc adm certificates regenerate-machine-config-server-serving-cert --update-ignition=false
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:398]: the command should be `oc adm ocp-certificates regenerate-machine-config-server-serving-cert --update-ignition=false`

File: modules/oc-adm-by-example-content.adoc [hunk 7]
```diff
@@ -365,16 +376,45 @@ Watch platform certificates.
...
+
+  # Update the user-data secrets to use new MCS certs
+  oc adm certificates update-ignition-ca-bundle-for-machine-config-server
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:401]: the command should be `oc adm ocp-certificates update-ignition-ca-bundle-for-machine-config-server`

File: modules/oc-adm-by-example-content.adoc [hunk 8]
```diff
@@ -365,16 +376,45 @@ Watch platform certificates.
...
 
 
 == oc adm ocp-certificates regenerate-top-level
 Regenerate the top level certificates in an OpenShift cluster
 
+.Example usage
+[source,bash,options="nowrap"]
+----
+  # Regenerate all known signing certificates on the cluster.
+  oc adm certificates regenerate-signers -A --all
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:413]: we don't have such command.

File: modules/oc-adm-by-example-content.adoc [hunk 9]
```diff
@@ -365,16 +376,45 @@ Watch platform certificates.
...
+
+  # Regenerate the signing certificate contained in a particular secret.
+  oc adm certificates regenerate-signers -n openshift-kube-apiserver-operator loadbalancer-serving-signer
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:416]: we don't have such command

File: modules/oc-adm-by-example-content.adoc [hunk 10]
```diff
@@ -384,6 +424,12 @@ Remove old CAs from ConfigMaps representing platform trust bundles in an OpenShi
 .Example usage
 [source,bash,options="nowrap"]
 ----
+  # Remove all known trust bundles in the cluster
+  oc adm ocp-certificates remove-old-trust configmaps -A --all
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:428]: the command will failed ,

File: modules/oc-adm-by-example-content.adoc [hunk 11]
```diff
@@ -384,6 +424,12 @@ Remove old CAs from ConfigMaps representing platform trust bundles in an OpenShi
...
+
+  # Remove a trust bundled contained in a particular config map
+  oc adm ocp-certificates remove-old-trust -n openshift-config-managed configmaps/kube-apiserver-aggregator-client-ca
```
Comment (zhouying7780) [inline modules/oc-adm-by-example-content.adoc:431]: the command will failed.

File: modules/oc-by-example-content.adoc (hunk 1 above)
Comment (bergerhoffer) [reply modules/oc-by-example-content.adoc:2212]: Opened https://issues.redhat.com/browse/OSDOCS-9586 to track updating references of "oc registry info" in the doc set to this alternative method.

**Comment by bergerhoffer** [comment]
Note to peer reviewer: this is autogenerated content, so I can't manually change the wording. But I do plan to submit fixes to the eng repos with a few fixes, so feel free to point out anything you notice.

File: modules/oc-adm-by-example-content.adoc [hunk 12]
```diff
@@ -729,6 +729,22 @@ Create a new OpenShift release
 == oc adm restart-kubelet
 Restarts kubelet on the specified nodes
 
+.Example usage
+[source,bash,options="nowrap"]
+----
+  # Restart all the nodes,  10% at a time
+  oc adm restart-kubelet nodes --all --directive=RemoveKubeletKubeconfig
```
Comment (mjpytlak) [inline modules/oc-adm-by-example-content.adoc:736]: All of the other examples have `--parallelism=` does this example require `--parallelism=10%`"?

File: modules/oc-adm-by-example-content.adoc [hunk 13]
```diff
@@ -729,6 +729,22 @@ Create a new OpenShift release
...
+
+  # Restart all the nodes,  20 nodes at a time
+  oc adm restart-kubelet nodes --all --parallelism=20 --directive=RemoveKubeletKubeconfig
```
Comment (mjpytlak) [inline modules/oc-adm-by-example-content.adoc:739]: Is --`parallelism=20` missing a `%`?

File: modules/oc-by-example-content.adoc [hunk 14]
```diff
@@ -1450,8 +1454,17 @@ Get documentation for a resource
   # Get the documentation of the resource and its fields
   oc explain pods
 
+  # Get all the fields in the resource
```
Comment (mjpytlak) [inline modules/oc-by-example-content.adoc:1457]:
```suggestion
  # Get all of the fields in the resource
```

File: modules/oc-by-example-content.adoc [hunk 15]
```diff
@@ -1544,12 +1557,27 @@ Display one or many resources
...
+== oc get-token
+Experimental: Get token from external OIDC issuer as credentials exec plugin
+
+.Example usage
+[source,bash,options="nowrap"]
+----
+  # Starts an auth code flow to the issuer url with the client id and the given extra scopes
+  oc get-token --client-id=client-id --issuer-url=test.issuer.url --extra-scopes=email,profile
+
+  # Starts an authe code flow to the issuer url with a different callback address.
```
Comment (mjpytlak) [inline modules/oc-by-example-content.adoc:1575]:
```suggestion
  # Starts an auth code flow to the issuer url with a different callback address
```

File: modules/oc-by-example-content.adoc [hunk 16]
```diff
@@ -1827,6 +1855,9 @@ Log in to a server
 
   # Log in to the given server through a browser
   oc login localhost:8443 --web --callback-port 8280
+
+  # Log in to the external OIDC issuer through Auth Code + PKCE by starting a local server listening port 8080
```
Comment (mjpytlak) [inline modules/oc-by-example-content.adoc:1859]:
```suggestion
  # Log in to the external OIDC issuer through Auth Code + PKCE by starting a local server listening on port 8080
```

File: modules/oc-by-example-content.adoc [hunk 17]
```diff
@@ -2999,12 +3018,15 @@ Experimental: Wait for a specific condition on one or many resources
...
 
-  # The default value of status condition is true; you can wait for other targets after an equal delimiter (compared after Unicode simple case folding, which is a more general form of case-insensitivity):
+  # The default value of status condition is true; you can wait for other targets after an equal delimiter (compared after Unicode simple case folding, which is a more general form of case-insensitivity)
   oc wait --for=condition=Ready=false pod/busybox1
 
-  # Wait for the pod "busybox1" to contain the status phase to be "Running".
+  # Wait for the pod "busybox1" to contain the status phase to be "Running"
   oc wait --for=jsonpath='{.status.phase}'=Running pod/busybox1
 
+  # Wait for the service "loadbalancer" to have ingress.
```
Comment (mjpytlak) [inline modules/oc-by-example-content.adoc:3027]:
```suggestion
  # Wait for the service "loadbalancer" to have ingress
```

**Review by mjpytlak — COMMENTED** [review]
@bergerhoffer Just a few odds and ends to look at. Thanks!

**Comment by bergerhoffer** [comment]
Thanks @mjpytlak! I'll add these items to my list of things to look into when I submit the fixes to the eng repos.

**Pull request merged**